crypto/src/kdf/mod.rs

```diff
@@ -93,10 +93,9 @@ pub enum KdfChallenge {
     },
 }
 
-fn make_salt<R: Rng + CryptoRng>(rng: &mut R, len: NonZeroUsize) -> Result<Vec<u8>, KdfError> {
-    let len = len.try_into().map_err(|_| KdfError::InvalidSaltSize)?;
-    let salt: Vec<u8> = (0..len).map(|_| rng.gen::<u8>()).collect();
-    Ok(salt)
+fn make_salt<R: Rng + CryptoRng>(rng: &mut R, len: NonZeroUsize) -> Vec<u8> {
+    let len = len.get();
+    (0..len).map(|_| rng.gen::<u8>()).collect()
```

Comment:
It seems like KdfError::InvalidSaltSize is no longer used, so it can be removed.